libbeat/processors/add_host_geo/add_host_geo.go

```diff
@@ -0,0 +1,59 @@
+package add_geo
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+func init() {
+	processors.RegisterPlugin("add_host_geo", newAddGeo)
+}
+
+func newAddGeo(c *common.Config) (processors.Processor, error) {
+	config := &Config{}
+	c.Unpack(config)
+
+	ag := addGeo{
+		"name":             config.Name,
+		"location":         config.Location,
+		"continent_name":   config.ContinentName,
+		"country_iso_code": config.CountryISOCode,
+		"region_name":      config.RegionName,
+		"city_name":        config.CityName,
+	}
+	// Delete any empty values
+	for k, v := range ag {
+		if v == "" {
+			delete(ag, k)
+		}
+	}
+
+	return ag, nil
+}
+
+type Config struct {
+	Name           string `config:"name"`
+	Location       string `config:"location"`
+	ContinentName  string `config:"continent_name"`
+	CountryISOCode string `config:"country_iso_code"`
+	RegionName     string `config:"region_name"`
+	CityName       string `config:"city_name"`
+}
+
+type addGeo common.MapStr
+
+func (ag addGeo) Run(event *beat.Event) (*beat.Event, error) {
+	_, err := event.PutValue("host.geo", ag.asMap())
```

Comment:
Does this really belong into the host? We are interested where the agent is running so it would be `agent.geo`?

Reply:
It's an interesting question. I agree that agent is slightly better, but I wonder if that is outweighed by the simplicity of not having yet another geo field. A location is tied to a host as much as it is tied to an agent.

There's an argument that's not true for the `geo.name` field we use, which is free-form, but perhaps we don't need to be so dogmatic there.

Reply:
One point that would speak for the host is that in the case of apm-server, apm-server is not the agent but the observer. There it would then be `observer.geo.*`. Having it under `host` would then not change it. I'm good with going with `host.*` for now and if someone is not happy with that, he can use the `rename` processor.

Reply:
+1 for **not** having `agent.geo.*`
As @ruflin says, in ECS, the `agent` is defined as running on a `host` or an `observer`.

Both `host.geo.*` and `observer.geo.*` are defined by ECS, so we are covered in either case.